access/src/main/java/zone/cogni/asquare/virtuoso/VirtuosoRdfStoreService.java

```diff
@@ -60,20 +46,13 @@ public VirtuosoRdfStoreService(String rdfStoreUrl, String rdfStoreUser, String r
     this.rdfStoreUser = rdfStoreUser;
     this.rdfStorePassword = rdfStorePassword;
     this.graphCrudUseBasicAuth = graphCrudUseBasicAuth;
+    //  TODO HttpClientBuilder httpClientBuilder = HttpClients.custom().useSystemProperties();
+    httpClient = httpClientBuilder(true).build();
+    queryExecutionBuilder = QueryExecutionHTTPBuilder.service(rdfStoreUrl).httpClient(httpClient);
   }
 
-  protected CloseableHttpClient buildHttpClient() {
-    HttpClientBuilder httpClientBuilder = HttpClients.custom().useSystemProperties();
-    httpClientBuilder.setConnectionManager(new PoolingHttpClientConnectionManager(60L, TimeUnit.SECONDS));
-    if (!StringUtils.isBlank(rdfStoreUser)) {
-      CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-      credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(rdfStoreUser, rdfStorePassword));
-      httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
-    }
-    else {
-      log.warn("Virtuoso executor service {} is configured without credentials.", rdfStoreUrl);
-    }
-    return httpClientBuilder.build();
+  private HttpClient.Builder httpClientBuilder(final boolean withAuthentication) {
+    return HttpClientUtils.createHttpClientBuilder(withAuthentication ? rdfStoreUser : null, rdfStorePassword).connectTimeout(Duration.ofSeconds(60));
```

Comment:
Ah true, thanks for pointing out. I missed that bit in the original code.

Will write some test for this too to avoid these kind of surprises ...

Reply:
So JDK 11 HttpClient [does not seem to support digest auth](https://bugs.openjdk.org/browse/JDK-8285888). The authentication challenge needs to be implemented from scratch ...

Reply:
@domivds currently reimplemented to generic SPARQL 1.1. protocol calls using Apache HttpClient (only the Virtuoso service) bypassing Jena query execution - in case it turns out not to be performant enough, HttpClient adjustments might be the way + added disabled integration tests for Virtuoso.

Reply:
OK, so rewired to `RDFConnection`, leaving out Apache HttpClient only in tests.